xdsl/irdl/declarative_assembly_format.py

```diff
@@ -908,22 +899,15 @@ def parse(self, parser: Parser, state: ParsingState) -> None:
         if unique_base is None:
             attr = parser.parse_attribute()
         elif self.unique_type is not None:
-            unique_base = cast(
-                type[TypedAttribute[Attribute]],
-                unique_base,
-            )
-            attr = unique_base.parse_with_type(parser, self.unique_type)
+            assert issubclass(unique_base, TypedAttribute)
+            attr = unique_base.parse_with_type(parser, self.unique_type)  # pyright: ignore[reportUnknownVariableType]
         elif issubclass(
             unique_base,
             ParametrizedAttribute,
         ):
             attr = unique_base.new(unique_base.parse_parameters(parser))
         elif issubclass(unique_base, Data):
-            unique_base = cast(
-                type[Data[Any]],
-                unique_base,
-            )
-            attr = unique_base.new(unique_base.parse_parameter(parser))
+            attr = unique_base.new(unique_base.parse_parameter(parser))  # pyright: ignore[reportUnknownVariableType]
```

Comment:
Not convinced telling pyright to ignore is better than an explicit cast?

Reply:
I now feel that telling pyright to ignore `Unknown` specifically is better, as a cast can become outdated with time, and we have no mechanisms to catch this.